src/util/address.rs

```diff
@@ -176,49 +382,207 @@ impl FromStr for AddressType {
     }
 }
 
+
+/// Version of the WitnessProgram reserved for the future use; subset of [`WitnessVersion`]
+/// corresponding number push opcodes `OP_PUSHNUM_2..=OP_PUSHNUM_16` having byte representation of
+/// `0x52..=0x60`.
+///
+/// Currently (after taproot softfork) includes all witness versions greater than 1.
+///
+/// Used in enumerations on Segwit info extratible from [`Address`] and [`TxOut`]/`scriptPubkey`s.
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
+#[repr(u8)]
+pub enum FutureWitnessVersion {
```

Comment:
So this type essentially guarantees next version will cause API break in this library. I see the appeal of being super-specific with types but this feels like too much.

Reply:
Not next version, but a future version after bitcoin softfork - which will be a major lib update and API break anyway.

Reply:
Yeah, that's what I meant. I don't think a new soft fork *has to be* API break. If there's something with the design of this library that causes it I'd like it to be changed.

Reply:
I do not get why we may like to have softfork as a non major version. I think we need to ensure that softforks are always major releases

Reply:
To not force people to change their code when they don't absolutely require it. It's already a massive pain to keep various libs in sync. Especially if you take into account MSRVs and even system dependencies.

Reply:
Accepting/supporting bitcoin softfork in any lib/project should be a conscious decision without defaulting to "auto-update"

Reply:
Auto-update does not make the code magically support a soft fork though.

Reply:
We should not require a major version bump for softforks. (Though in practice there is a good chance we will.)

Reply:
Removed the type